src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -0,0 +1,222 @@
+using System;
+using System.Buffers;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Text;
+using Uno.Disposables;
+
+namespace Windows.Storage.Streams
+{
+	public partial class DataReader : IDataReader, IDisposable
+	{
+		private readonly static ArrayPool<byte> _pool = ArrayPool<byte>.Create();
+		private readonly byte[] _bytes;
+
+		private int _bufferPosition = 0;
+
+		private DataReader(IBuffer buffer)
+		{
+			_bytes = buffer.ToArray();
+		}
+
+		public static DataReader FromBuffer(IBuffer buffer)
+		{
+			return new DataReader(buffer);
+		}
+
+		public UnicodeEncoding UnicodeEncoding { get; set; }
+
+		public ByteOrder ByteOrder { get; set; }
+
+		public uint UnconsumedBufferLength => (uint)(_bytes.Length - _bufferPosition);
+
+		public byte ReadByte()
+		{
+			VerifyRead(1);
+			var value = _bytes[_bufferPosition];
+			_bufferPosition++;
+			return value;
+		}
+
+		public void ReadBytes(byte[] value)
+		{
+			VerifyRead(value.Length);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'value' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)